api/src/main/java/org/ehcache/spi/Hookable.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.spi;
+
+/**
+ * @author Abhilash
+ *
+ */
+public interface Hookable {
```

Comment:
And it probably should extend `UserManagedCache` as well. This will enable to get rid of a large number of the casts introduced in various places.